### ISSUE TITLE
Consistency issues.

### ISSUE THREAD

**Issue by mrinc** [bchavez/RethinkDb.Driver#24]
```
while (true) {
  Console.Write("START");
  var _r = RethinkDB.r;
  var conn = _r.connection()
      .hostname("IP")
      .port(RethinkDBConstants.DEFAULT_PORT)
      .timeout(60)
      .connect();
  var x = _r.db("DB")
      .table("TABLE")
      .count();
  long resCount = x.run(conn);
  Console.Write(" - C: " + resCount);
  conn.close();
  conn = null;
  _r = null;
  Console.WriteLine("FINISH");
  Console.WriteLine();
  Console.WriteLine();
}
```

So with that code I keep getting 
```
"An unhandled exception of type 'System.AggregateException' occurred in mscorlib.dll
Additional information: One or more errors occurred."
Inner: 
Object reference not set to an instance of an object.
"   at RethinkDb.Driver.Net.SocketWrapper.SendQuery(Int64 token, String json, Boolean awaitResponse)
   at RethinkDb.Driver.Net.Connection.SendQuery(Query query, Boolean awaitResponse)
   at RethinkDb.Driver.Net.Connection.<RunQueryAsync>d__30`1.MoveNext()"
```
Any ideas why?

Thanks

**Comment by bchavez** [comment]
Hi there. What version are you using? CoreCLR or Full?

**Comment by bchavez** [comment]
I created a unit test with your code, and I'm not seeing any `AggregateException` on an empty table.

**Comment by mrinc** [comment]
Hey,

Thanks for the quick reply.

I installed the ReThinkDb.Driver v2.2.3

Not sure which one is included in the nuget package.

Thanks

**Comment by bchavez** [comment]
I cant seem to reproduce the error. Do you have a sample project I could use to test?

**Comment by mrinc** [comment]
Uploaded csproj - https://onedrive.live.com/redir?resid=4581A44EC3B61ED9!351940&authkey=!AB0RzlgUXltaI4A&ithint=file%2czip

Could it be my server is out of date or something?

**Comment by bchavez** [comment]
@mrinc , thanks so much for your sample project. I'm able to reproduce the issue. Seems to be concurrency related with using TPL. Give me a few to investigate. I'll keep you updated.

**Comment by bchavez** [comment]
Hey @mrinc , thanks for reporting this. This one was a tricky bug. Basically, we we're late setting a `CancellationToken` in that tight loop which caused the null reference exception.

This should be fixed in **`2.2.4`** Full .NET Framework and **`2.2.4-beta-1`** for CoreCLR.

Please let me know if it fixes your issue.

**Comment by mrinc** [comment]
A+

All looks good on my side :)

Thanks for the help!
